File: CS207b/L/w02/src_fix/SortedList.py
from sort_algos.SortStrat import SortStrat


class SortedList:
    algo = None

    def __init__(self, algo_name):
        self.algo = SortStrat(algo_name)

    def sort(self, arr):
        return self.algo.sort(arr)

    def print(self):
        print(self.algo.arr)

File: CS207b/L/w02/src_go/sortalgos/sortedlist.go
package sortalgos

// Context

type SortedList struct {
	Data []int
	Algo SortAlgo
}

func InitSL(data []int, algo SortAlgo) *SortedList {
	return &SortedList{
		Data: data,
		Algo: algo,
	}
}

func (sl *SortedList) SetAlgo(algo SortAlgo) {
	sl.Algo = algo
}

func (sl *SortedList) Sort() []int {
	return sl.Algo.Sort(sl.Data)
}
